userspace/libsinsp/filterchecks.cpp

```diff
@@ -109,6 +110,317 @@ static inline bool str_match_start(const std::string& val, size_t len, const cha
+
+	m_path_checks.emplace(PPME_SYSCALL_QUOTACTL_X, evt_arg_special);
+	m_success_checks.emplace(PPME_SYSCALL_QUOTACTL_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_RENAME_X, evt_arg_oldpath);
+	m_target_checks.emplace(PPME_SYSCALL_RENAME_X, evt_arg_newpath);
+	m_success_checks.emplace(PPME_SYSCALL_RENAME_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_RENAMEAT_X, evt_arg_oldpath);
+	m_target_checks.emplace(PPME_SYSCALL_RENAMEAT_X, evt_arg_newpath);
+	m_success_checks.emplace(PPME_SYSCALL_RENAMEAT_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_RENAMEAT2_X, evt_arg_oldpath);
+	m_target_checks.emplace(PPME_SYSCALL_RENAMEAT2_X, evt_arg_newpath);
+	m_success_checks.emplace(PPME_SYSCALL_RENAMEAT2_X, evt_arg_res_eq_0);
+
+	m_success_checks.emplace(PPME_SYSCALL_LINK_X, evt_arg_res_eq_0);
+
+	m_success_checks.emplace(PPME_SYSCALL_LINKAT_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_LINK_2_X, evt_arg_newpath);
+	m_target_checks.emplace(PPME_SYSCALL_LINK_2_X, evt_arg_oldpath);
+	m_success_checks.emplace(PPME_SYSCALL_LINK_2_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_LINKAT_2_X, evt_arg_newpath);
+	m_target_checks.emplace(PPME_SYSCALL_LINKAT_2_X, evt_arg_oldpath);
+	m_success_checks.emplace(PPME_SYSCALL_LINKAT_2_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_SYMLINK_X, evt_arg_linkpath);
+	m_target_checks.emplace(PPME_SYSCALL_SYMLINK_X, evt_arg_target);
+	m_success_checks.emplace(PPME_SYSCALL_SYMLINK_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_SYMLINKAT_X, evt_arg_linkpath);
+	m_target_checks.emplace(PPME_SYSCALL_SYMLINKAT_X, evt_arg_target);
+	m_success_checks.emplace(PPME_SYSCALL_SYMLINKAT_X, evt_arg_res_eq_0);
+
+	m_source_checks.emplace(PPME_SYSCALL_MOUNT_X, evt_arg_dev);
+	m_target_checks.emplace(PPME_SYSCALL_MOUNT_X, evt_arg_dir);
+	m_success_checks.emplace(PPME_SYSCALL_MOUNT_X, evt_arg_res_eq_0);
+
+	m_path_checks.emplace(PPME_SYSCALL_UMOUNT_X, evt_arg_name);
+	m_success_checks.emplace(PPME_SYSCALL_UMOUNT_X, evt_arg_res_eq_0);
+
+	m_path_checks.emplace(PPME_SYSCALL_UMOUNT_1_X, evt_arg_name);
+	m_success_checks.emplace(PPME_SYSCALL_UMOUNT_1_X, evt_arg_res_eq_0);
+
+	m_path_checks.emplace(PPME_SYSCALL_UMOUNT2_X, evt_arg_name);
+	m_success_checks.emplace(PPME_SYSCALL_UMOUNT2_X, evt_arg_res_eq_0);
+}
+
+sinsp_filter_check* sinsp_filter_check_fspath::allocate_new()
+{
+	return (sinsp_filter_check*)new sinsp_filter_check_fspath();
+}
+
+uint8_t* sinsp_filter_check_fspath::extract(sinsp_evt* evt, OUT uint32_t* len, bool sanitize_strings)
+{
+	*len = 0;
+	ASSERT(evt);
+
+	m_tstr = "";
+
+	if(m_path_checks.empty())
+	{
+		create_fspath_checks();
+	}
+
+	// First check the success conditions.
+	auto it = m_success_checks.find(evt->get_type());
+	if(it == m_success_checks.end())
+	{
+		return NULL;
+	}
+
+	if(!it->second->compare(evt))
+	{
+		return NULL;
+	}
+
+	std::optional<std::string> enter_param;
```

Comment:
This will create a copy. Can't we have an optional reference wrapper?

Reply:
Yes, replaced with std::optional<std::reference_wrapper<std::string>>.